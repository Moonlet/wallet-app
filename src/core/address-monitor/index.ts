import { Notifications } from '../messaging/notifications/notifications';
import { Blockchain } from '../blockchain/types';
import DeviceInfo from 'react-native-device-info';
import { IWalletsState, IAccountState } from '../../redux/wallets/state';
import { CONFIG } from '../../config';

const url = CONFIG.dataApiUrl + '/notifications/register';
const monitoredBlockchains = [Blockchain.ETHEREUM, Blockchain.ZILLIQA];

// const getDeviceToken = (): Promise<string> =>
//     Platform.OS === 'android' ? Notifications.getToken() : getApnsToken();

export const updateAddressMonitorTokens = (wallets: IWalletsState) => {
    const addresses = {};

    Object.keys(wallets || []).forEach(id => {
        wallets[id].accounts.forEach((account: IAccountState) => {
            if (monitoredBlockchains.indexOf(account.blockchain) !== -1) {
                if (!addresses[account.blockchain]) {
                    addresses[account.blockchain] = [];
                }

                addresses[account.blockchain].push(account.address);
            }
        });
    });

    const requestAddresses = Object.keys(addresses).map(blockchain => ({
        blockchain,
        addresses: addresses[blockchain]
    }));

    const deviceId = DeviceInfo.getUniqueId();
    Notifications.getToken().then(token => {
        const request = {
            method: 'POST',
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify({
                deviceId,
                tokenType: 'fcm',
                token,
                addresses: requestAddresses
            })
        };

        fetch(url, request);
    });
};
