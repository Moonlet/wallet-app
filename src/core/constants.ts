import { Blockchain } from './blockchain/types';

const BLOCKCHAIN_COINS = {
    [Blockchain.ETHEREUM]: 'ETH',
    [Blockchain.ZILLIQA]: 'ZIL',
    [Blockchain.COSMOS]: 'ATOM',
    [Blockchain.STELLAR]: 'XLM'
};

export { BLOCKCHAIN_COINS };
