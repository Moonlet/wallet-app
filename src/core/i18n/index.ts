export * from './translation/translate-component';
export * from './translation/translate';
