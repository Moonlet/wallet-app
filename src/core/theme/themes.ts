export enum Theme {
    dark = 'dark',
    light = 'light'
}
