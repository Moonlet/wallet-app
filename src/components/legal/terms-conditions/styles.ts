import { StyleSheet, Platform } from 'react-native';
import { ITheme } from '../../../core/theme/itheme';
import { BASE_DIMENSION, ICON_CONTAINER_SIZE } from '../../../styles/dimensions';

export default (theme: ITheme) =>
    StyleSheet.create({
        container: {
            flex: 1,
            paddingTop: Platform.OS === 'ios' ? BASE_DIMENSION * 5 : 0,
            justifyContent: 'center',
            backgroundColor: theme.colors.appBackground
        },
        button: {
            flexDirection: 'row',
            alignItems: 'center'
        },
        iconContainer: {
            width: ICON_CONTAINER_SIZE,
            height: ICON_CONTAINER_SIZE,
            alignItems: 'center',
            justifyContent: 'center',
            marginLeft: BASE_DIMENSION
        },
        icon: {
            color: theme.colors.accent
        },
        loadingContainer: {
            flex: 1,
            overflow: 'hidden',
            backgroundColor: theme.colors.appBackground
        },
        loadingIndicator: {
            position: 'absolute',
            top: 0,
            bottom: 0,
            left: 0,
            right: 0,
            backgroundColor: theme.colors.appBackground
        }
    });
