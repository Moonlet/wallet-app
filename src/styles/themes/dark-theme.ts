import { ITheme } from '../../core/theme/itheme';
import { COLORS } from '../colors';

export const darkTheme: ITheme = {
    dark: true,
    shadowGradient: ['#00000000', '#000000CC'],

    fontSize: {
        small: 12,
        regular: 16,
        large: 34
    },

    colors: {
        primary: COLORS.GRAY,
        accent: COLORS.AQUA,
        text: COLORS.WHITE,
        textDarker: COLORS.LIGHT_GRAY,
        positive: COLORS.HOT_GREEN,
        negative: COLORS.HOT_RED,

        cardBackground: COLORS.DARK_GRAY,
        appBackground: COLORS.COD_GRAY,
        settingsDivider: COLORS.BLACK
    }
};
