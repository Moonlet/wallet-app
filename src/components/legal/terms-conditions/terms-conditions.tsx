import React from 'react';
import { View, TouchableOpacity, Platform } from 'react-native';
import { withTheme, IThemeProps } from '../../../core/theme/with-theme';
import stylesProvider from './styles';
import { smartConnect } from '../../../core/utils/smart-connect';
import Icon from '../../icon';
import { ICON_SIZE, BASE_DIMENSION } from '../../../styles/dimensions';
import { LoadingIndicator } from '../../loading-indicator/loading-indicator';
import WebView from '../../../library/webview/webview';

interface IExternalProps {
    onClose?: () => void;
    showClose?: boolean;
}

export const TCComponent = (
    props: IExternalProps & IThemeProps<ReturnType<typeof stylesProvider>>
) => {
    return (
        <View
            style={[
                props.styles.container,
                { paddingTop: Platform.OS === 'ios' && props.showClose ? BASE_DIMENSION * 5 : 0 }
            ]}
        >
            <React.Fragment>
                {props.showClose && (
                    <TouchableOpacity
                        onPress={() => props.onClose && props.onClose()}
                        style={props.styles.button}
                    >
                        <View style={props.styles.iconContainer}>
                            <Icon name="close" size={ICON_SIZE} style={props.styles.icon} />
                        </View>
                    </TouchableOpacity>
                )}
                <View style={props.styles.webviewContainer}>
                    <WebView
                        startInLoadingState={true}
                        source={{ uri: 'https://moonlet.xyz/terms-and-conditions/' }}
                        renderLoading={() => (
                            <View style={props.styles.loadingIndicator}>
                                <LoadingIndicator />
                            </View>
                        )}
                    />
                </View>
            </React.Fragment>
        </View>
    );
};

export const TC = smartConnect<IExternalProps>(TCComponent, [withTheme(stylesProvider)]);
