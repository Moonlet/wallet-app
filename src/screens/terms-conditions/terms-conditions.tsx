import React from 'react';
import { View } from 'react-native';
import { withTheme, IThemeProps } from '../../core/theme/with-theme';
import { translate } from '../../core/i18n';
import stylesProvider from './styles';
import { smartConnect } from '../../core/utils/smart-connect';
import { TC } from '../../components/legal/terms-conditions/terms-conditions';

export const navigationOptions = () => ({
    title: translate('App.labels.tc')
});

export const TermsConditionsScreenComponent = (
    props: IThemeProps<ReturnType<typeof stylesProvider>>
) => {
    return (
        <View style={props.styles.container}>
            <TC />
        </View>
    );
};

TermsConditionsScreenComponent.navigationOptions = navigationOptions;

export const TermsConditionsScreen = smartConnect(TermsConditionsScreenComponent, [
    withTheme(stylesProvider)
]);
