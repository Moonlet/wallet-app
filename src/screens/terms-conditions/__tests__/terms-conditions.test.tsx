import React from 'react';
import { TermsConditionsScreenComponent, IProps } from '../terms-conditions';
import { darkTheme } from '../../../styles/themes/dark-theme';
import styleProvider from '../styles';

import { shallow } from 'enzyme';
import { loadTranslations } from '../../../core/i18n';

const props: IProps = {
    // @ts-ignore
    navigation: {
        navigate: jest.fn()
    },
    styles: styleProvider(darkTheme)
};

describe('terms and conditions screen component', () => {
    beforeAll(async () => {
        await loadTranslations('en');
    });

    it('renders correctly', () => {
        const wrapper = shallow(<TermsConditionsScreenComponent {...props} />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
