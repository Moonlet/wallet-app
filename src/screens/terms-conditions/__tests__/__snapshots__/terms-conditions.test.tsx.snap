// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`terms and conditions screen component renders correctly 1`] = `
"<View style={{...}}>
  <Image style={{...}} source={{...}} />
  <withTheme(TextComponent) style={{...}}>
    Launching soon!
  </withTheme(TextComponent)>
</View>"
`;
